### ISSUE TITLE
Max - dynamic lua is broken for floats and range

### ISSUE THREAD

**Issue by smallg2020** [TheGameCreators/GameGuruRepo#2617]
using the float symbol or the range setting doesn't seem to be working in the new versions of max
a simple script like this should let you choose a float to display from the editor but it doesn't respect the range set (range is always 0 - 100) and the number shown in test game is not what you set / see in the editor
floats.lua
`--DESCRIPTION: [MYFLOAT#=0.5(0,1)]

g_floats = {}
function floats_init(e)
	g_floats[e] = {}
	floats_properties(e, 0.5)
end 

function floats_properties(e, myfloat)
	g_floats[e].myfloat = myfloat 
end 

function floats_main(e)
	Prompt(g_floats[e].myfloat)	
end `

**Comment by marcuswilm** [comment]
Hello, yesterday I posted a comment in this thread https://forum.game-guru.com/thread/223409 posted by smallg.

Personally I would like the avaibility of the float numbers in GameGuru MAX for some of my scripts.

Thank you.

**Comment by AmenMoses** [comment]
Something else that is missing in dynamic Lua is the ability to mark variables as pertaining to all entities using the script as opposed to being entity specific.

**Comment by LeeBamberTGC** [comment]
Fixed for next EXP build.

**Comment by marcuswilm** [comment]
I tested the float numbers in dynamic lua in the latest alpha version (2022.04.29) with the script (behavior "test_float_numbers.lua") included in the attached file (.lua to .txt extension).

[test_float_numbers.txt](https://github.com/TheGameCreators/GameGuruRepo/files/8609680/test_float_numbers.txt)

In my quality of simple alpha tester I cannot guess how far the accuracy of the float numbers is defined in GameGuru MAX dynamic Lua,  
That's why it is up to the GG MAX developers staff to confirm if there is still a problem or not.

In his comment, smallg2020, mentioned this : "it doesn't respect the range set ". It is indeed a problem that I noticed since a long time.

As you know, when you are testing some behavior (in my case, with many trials and errors for the most complex ones), I regularly save them in my custom folder and I reload the 'game' and 'level' to be sure that the behavior is updated. (I even delete the behavior in my level before saving it).
And surprisingly when I assign again the behavior in the level, I see 'ghost' values from previous tests that appear in the input fields of that behavior. Therefore I may guess that there is somewhere in the GGMAX folders some 'cache' retaining those previous values.
It doesn't help for debugging custom behaviors.

:)
